From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Nicolas Thill <nico@openwrt.org>
Date: Fri, 9 Jan 2015 21:04:35 +0100
Subject: [PATCH] Makefile.tools: link the readline users against ncurses

The tools that link -lreadline need -lncurses spelled out next to it to
link on OpenWrt.
---
 Makefile.tools | 18 +++++++++---------
 1 file changed, 9 insertions(+), 9 deletions(-)

--- a/Makefile.tools
+++ b/Makefile.tools
@@ -19,7 +19,7 @@ client_bluetoothctl_SOURCES = client/mai
 					client/telephony.h client/telephony.c
 client_bluetoothctl_LDADD = lib/libbluetooth-internal.la \
 			gdbus/libgdbus-internal.la src/libshared-glib.la \
-			$(GLIB_LIBS) $(DBUS_LIBS) -lreadline
+			$(GLIB_LIBS) $(DBUS_LIBS) -lreadline -lncurses
 endif
 
 if ZSH_COMPLETIONS
@@ -388,7 +388,7 @@ tools_meshctl_SOURCES = tools/meshctl.c
 				tools/mesh-gatt/onoff-model.c
 tools_meshctl_LDADD = gdbus/libgdbus-internal.la src/libshared-glib.la \
 				lib/libbluetooth-internal.la \
-				$(GLIB_LIBS) $(DBUS_LIBS) -ljson-c -lreadline
+				$(GLIB_LIBS) $(DBUS_LIBS) -ljson-c -lreadline -lncurses
 endif
 
 bin_PROGRAMS +=  tools/mesh-cfgclient
@@ -405,7 +405,7 @@ tools_mesh_cfgclient_SOURCES = tools/mes
 				mesh/crypto.h mesh/crypto.c
 
 tools_mesh_cfgclient_LDADD = lib/libbluetooth-internal.la src/libshared-ell.la \
-						$(ell_ldadd) -ljson-c -lreadline
+						$(ell_ldadd) -ljson-c -lreadline -lncurses
 
 bin_PROGRAMS +=  tools/mesh-cfgtest
 
@@ -487,7 +487,7 @@ noinst_PROGRAMS += tools/btmgmt tools/ob
 tools_obex_client_tool_SOURCES = $(gobex_sources) $(btio_sources) \
 						tools/obex-client-tool.c
 tools_obex_client_tool_LDADD = lib/libbluetooth-internal.la \
-			       src/libshared-glib.la $(GLIB_LIBS) -lreadline
+			       src/libshared-glib.la $(GLIB_LIBS) -lreadline -lncurses
 
 tools_obex_server_tool_SOURCES = $(gobex_sources) $(btio_sources) \
 						tools/obex-server-tool.c
@@ -498,16 +498,16 @@ tools_bluetooth_player_SOURCES = tools/b
 				client/player.c
 tools_bluetooth_player_LDADD = gdbus/libgdbus-internal.la \
 				src/libshared-glib.la \
-				$(GLIB_LIBS) $(DBUS_LIBS) -lreadline
+				$(GLIB_LIBS) $(DBUS_LIBS) -lreadline -lncurses
 
 tools_obexctl_SOURCES = tools/obexctl.c
 tools_obexctl_LDADD = gdbus/libgdbus-internal.la src/libshared-glib.la \
-			$(GLIB_LIBS) $(DBUS_LIBS) -lreadline
+			$(GLIB_LIBS) $(DBUS_LIBS) -lreadline -lncurses
 
 tools_btmgmt_SOURCES = tools/btmgmt.c src/uuid-helper.c client/display.c \
 			client/mgmt.c
 tools_btmgmt_LDADD = lib/libbluetooth-internal.la src/libshared-glib.la \
-				$(GLIB_LIBS) -lreadline
+				$(GLIB_LIBS) -lreadline -lncurses
 if DEPRECATED
 noinst_PROGRAMS += attrib/gatttool
 
@@ -517,7 +517,7 @@ attrib_gatttool_SOURCES = attrib/gatttoo
 				attrib/utils.c src/log.c client/display.c \
 				client/display.h
 attrib_gatttool_LDADD = lib/libbluetooth-internal.la \
-			src/libshared-glib.la $(GLIB_LIBS) -lreadline
+			src/libshared-glib.la $(GLIB_LIBS) -lreadline -lncurses
 
 endif
 endif
@@ -573,5 +573,5 @@ client/btpclient/btpclient.$(OBJEXT): sr
 
 client_btpclient_btpclientctl_SOURCES = client/btpclient/btpclientctl.c client/display.c
 client_btpclient_btpclientctl_LDADD = src/libshared-mainloop.la src/libshared-glib.la \
-				lib/libbluetooth-internal.la -lreadline
+				lib/libbluetooth-internal.la -lreadline -lncurses
 endif
